plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveStorageFormat.java

```diff
@@ -160,6 +165,14 @@ else if (type.getCategory() == Category.PRIMITIVE) {
         }
     }
 
+    private static final Map<StorageFormat, HiveStorageFormat> HIVE_STORAGE_FORMAT_FROM_STORAGE_FORMAT = Arrays.stream(HiveStorageFormat.values())
+            .collect(toImmutableMap(format -> StorageFormat.create(format.getSerDe(), format.getInputFormat(), format.getOutputFormat()), identity()));
+
+    public static HiveStorageFormat getHiveStorageFormat(StorageFormat storageFormat)
+    {
+        return requireNonNull(HIVE_STORAGE_FORMAT_FROM_STORAGE_FORMAT.get(storageFormat), "Could not find format for " + storageFormat);
```

Comment:
Tables can have arbitrary formats, so if the mapping fails, we should assume they are index based.

Since this is for reading, I think we can ignore output format here (just use input format and serde).

Reply:
I changed the method return to be Optional<HiveStorageFormat>, and changed the map key to be a class with just the serDe and inputFormat.